src/full/Agda/Termination/TermCheck.hs

```diff
@@ -502,7 +502,8 @@ termDef name = terSetCurrent name $ inConcreteOrAbstractMode name $ \ def -> do
   -- which are not of data or record type.
 
   withoutKEnabled <- liftTCM withoutKOption
-  applyWhen withoutKEnabled (setMasks t) $ do
+  safeEnabled <- liftTCM safeOption
+  applyWhen (withoutKEnabled && not safeEnabled) (setMasks t) $ do
```

Comment:
I agree with @nad, `--safe` should not be used to determine the flavor of the Agda language.  If a flag is needed to enable this relaxation, it probably should be a new flag.  But also, see @nad's comment that excluding `postulate`s may actually not be sufficient.